fml/platform/darwin/scoped_nsobject.h

```diff
@@ -37,29 +37,42 @@ namespace fml {
 // We check for bad uses of scoped_nsobject and NSAutoreleasePool at compile
 // time with a template specialization (see below).
 
+class scoped_nsprotocol_arc_memory_management {
```

Comment:
Been a while since I've done heavy C++, but can this just be a namespace?

Reply:
Yep, it could be.  The reason things are a laid out the way they are is to keep the code somewhat similar to the chrome implementation.  I'll give it another style pass once I can verify that the generated machine code is fine.